src/genie_python/genie_epics_api.py

```diff
@@ -694,13 +701,13 @@ def check_alarms(self, blocks: list[str]) -> tuple[list[str], list[str], list[st
         Returns:
             list, list, list: the blocks in minor, major and invalid alarm
         """
-        alarm_states = self._get_fields_from_blocks(blocks, "SEVR", "alarm state")
+        alarm_states = self._get_fields_from_blocks(list(blocks), "SEVR", "alarm state")
         minor = [t[0] for t in alarm_states if t[1] == "MINOR"]
         major = [t[0] for t in alarm_states if t[1] == "MAJOR"]
         invalid = [t[0] for t in alarm_states if t[1] == "INVALID"]
         return minor, major, invalid
 
-    def check_limit_violations(self, blocks: list[str]) -> list[str]:
+    def check_limit_violations(self, blocks: typing.Tuple[str, ...]) -> list[str]:
```

Comment:
Type hint says tuple, comment says list. Which is it?

If you don't care, then the correct answer might be `Iterable[str]`